staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go

```diff
@@ -285,6 +285,16 @@ type ValidationRule struct {
 	// e.g. for attribute `foo.34$` appears in a list `testList`, the fieldPath could be set to `.testList['foo.34$']`
 	// +optional
 	FieldPath string `json:"fieldPath,omitempty" protobuf:"bytes,5,opt,name=fieldPath"`
+
+	// optionalOldSelf is used to indicate a transition rule is always evaluated.
+	// If enabled:
+	// - During a create, `oldSelf` evaluates to `null`
+	// - During an update, `oldSelf` evaluates to `null` when the old object is
+	//	 missing the field or if the field is uncorrelatable.
+	// If `oldSelf` is not used in `rule`, this field is ignored.
+	// +featureGate=CRDValidationRatcheting
+	// +optional
+	OptionalOldSelf *bool `json:"optionalOldSelf,omitempty" protobuf:"bytes,6,opt,name=optionalOldSelf"`
```

Comment:
I like the approach, and the polarity (absent / false keeps behavior as it is today, present+true opts into evaluating even when oldSelf is null).

It wouldn't be an API review without name bikeshedding :-D

stating what we're trying to communicate a few different ways:
* oldSelf is optional
* allow oldSelf to be null
* evaluate [even] when oldSelf is null
* evaluate [even] with null oldSelf

reflections on those:
* "allow" is overloaded with policy/authz; "evaluate" seems clearer to me
* I like including null in the name somehow if that's what their expression would need to check for explicitly (is it? see questions above about type checking and null handling); it seems clearer to me than "optional"
* avoiding a name that leads double-negatives would be ideal (`optionalOldSelf: false` feels like a double negative to me for some reason)

Something like `evaluateWithNullOldSelf: true` seems clearer than `optionalOldSelf: true`, but maybe I've been staring at this for too long.

There's also our [historical avoidance of booleans](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types) in favor of enum-like policies

> Think twice about bool fields. Many ideas start as boolean but eventually trend towards a small set of mutually exclusive options. Plan for future expansions by describing the policy options explicitly as a string type alias (e.g. TerminationMessagePolicy).

Not sure that applies here, but worth at least considering (e.g. `oldSelfEvaluationPolicy: "WhenPresent" | "Always"`)

Reply:
I could get behind an enum like `oldSelfEvaluationPolicy` even if we never end up using more cases...

Spitballing some alternative names and cases none of which i totally like hoping we can reach a consensus soon:

1. oldSelfEvaluationPolicy: "WhenPresent" (default) | "Always" (from Jordans comment above)
2. oldSelfBehavior: "Transitions" (default) | Always
3. transitionRulePolicy: "Existing" |  "Unconditional"

/cc @jpbetz @cici37 

for your takes

Reply:
I agree that `optionalOldSelf` sounds less clear especially with false value comparing with the suggestion raised. 
I am slightly leaning towards `oldSelfEvaluationPolicy: "WhenPresent" | "Always"`(transitions is a term user has to look up to know exactly what it refers to here, and we already use `Always` other places comparing with `Unconditional` so keep it consistent would be great) so that user knows exactly what they signed up for and plus leave room for future possibility.

Reply:
with the approach of making `oldSelf` an actual optional value, and getting typechecking to require the rule to treat it that way, I think `optionalOldSelf bool` is more reasonable